pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -145,12 +145,50 @@ let
       KEY_DH_OPERATIONS = whenAtLeast "4.7" yes;
 
       # needed for nftables
+      NF_LOG_NETDEV               = whenAtLeast "5.2" module;
+      NF_CT_NETLINK_HELPER        = whenAtLeast "5.2" module;
+      NETFILTER_NETLINK_GLUE_CT   = whenAtLeast "5.2" yes;
+      NF_TABLES_SET               = whenAtLeast "5.2" module;
       NF_TABLES_INET              = whenAtLeast "4.17" yes;
       NF_TABLES_NETDEV            = whenAtLeast "4.17" yes;
+      NFT_NUMGEN                  = whenAtLeast "5.2" module;
+      NFT_FLOW_OFFLOAD            = whenAtLeast "5.2" module;
+      NFT_CONNLIMIT               = whenAtLeast "5.2" module;
+      NFT_NAT                     = whenAtLeast "5.2" module;
+      NFT_TUNNEL                  = whenAtLeast "5.2" module;
+      NFT_OBJREF                  = whenAtLeast "5.2" module;
+      NFT_QUOTA                   = whenAtLeast "5.2" module;
+      NFT_REJECT_INET             = whenAtLeast "5.2" module;
+      NFT_FIB                     = whenAtLeast "5.2" module;
+      NFT_FIB_INET                = whenAtLeast "5.2" module;
+      NFT_XFRM                    = whenAtLeast "5.2" module;
+      NFT_SOCKET                  = whenAtLeast "5.2" module;
+      NFT_OSF                     = whenAtLeast "5.2" module;
+      NFT_TPROXY                  = whenAtLeast "5.2" module;
+      NFT_SYNPROXY                = whenAtLeast "5.3" module;
+      NF_DUP_NETDEV               = whenAtLeast "5.2" module;
+      NFT_DUP_NETDEV              = whenAtLeast "5.2" module;
+      NFT_FWD_NETDEV              = whenAtLeast "5.2" module;
+      NFT_FIB_NETDEV              = whenAtLeast "5.2" module;
+      NF_FLOW_TABLE_INET          = whenAtLeast "5.2" module;
+      NF_FLOW_TABLE               = whenAtLeast "5.2" module;
+      NETFILTER_XT_MATCH_IPCOMP   = whenAtLeast "5.2" module;
       NF_TABLES_IPV4              = whenAtLeast "4.17" yes;
+      NFT_REJECT_IPV4             = whenAtLeast "5.2" module;
+      NFT_DUP_IPV4                = whenAtLeast "5.2" module;
+      NFT_FIB_IPV4                = whenAtLeast "5.2" module;
       NF_TABLES_ARP               = whenAtLeast "4.17" yes;
+      NF_FLOW_TABLE_IPV4          = whenAtLeast "5.2" module;
+      NF_LOG_ARP                  = whenAtLeast "5.2" module;
```

Comment:
Have you checked if any options are not already enabled by default:
i.e. this one: https://cateee.net/lkddb/web-lkddb/NF_LOG_ARP.html
Is there any option you want that was not enabled as a module?

Reply:
First saved default.config, then added the nftables options.
`diff -Nuar default.config default_nft.config`

I suggested that a full replacement iptables firewall requires all the features of nftables.

Rechecked PR.

Reply:
Checked with zcat /proc/config.gz in running system
`diff -Nuar ~/kernel_config_1.txt ~/kernel_config_2.txt`
```
--- /home/lf/kernel_config_1.txt        2019-10-14 19:27:38.759418581 +0300
+++ /home/lf/kernel_config_2.txt        2019-10-14 17:53:32.799282574 +0300
@@ -1085,11 +1085,11 @@
 CONFIG_NETFILTER_CONNCOUNT=m
 CONFIG_NF_CONNTRACK_MARK=y
 CONFIG_NF_CONNTRACK_SECMARK=y
-# CONFIG_NF_CONNTRACK_ZONES is not set
+CONFIG_NF_CONNTRACK_ZONES=y
 CONFIG_NF_CONNTRACK_PROCFS=y
-# CONFIG_NF_CONNTRACK_EVENTS is not set
-# CONFIG_NF_CONNTRACK_TIMEOUT is not set
-# CONFIG_NF_CONNTRACK_TIMESTAMP is not set
+CONFIG_NF_CONNTRACK_EVENTS=y
+CONFIG_NF_CONNTRACK_TIMEOUT=y
+CONFIG_NF_CONNTRACK_TIMESTAMP=y
 CONFIG_NF_CONNTRACK_LABELS=y
 CONFIG_NF_CT_PROTO_DCCP=y
 CONFIG_NF_CT_PROTO_GRE=y
@@ -1107,7 +1107,9 @@
 CONFIG_NF_CONNTRACK_SIP=m
 CONFIG_NF_CONNTRACK_TFTP=m
 CONFIG_NF_CT_NETLINK=m
-# CONFIG_NETFILTER_NETLINK_GLUE_CT is not set
+CONFIG_NF_CT_NETLINK_TIMEOUT=m
+CONFIG_NF_CT_NETLINK_HELPER=m
+CONFIG_NETFILTER_NETLINK_GLUE_CT=y
 CONFIG_NF_NAT=m
 CONFIG_NF_NAT_AMANDA=m
 CONFIG_NF_NAT_FTP=m
```
The first variant is incorrect?

Reply:
Optimized options